components/publish/PublishWidget.vue

```diff
@@ -64,7 +64,18 @@ const { editor } = useTiptap({
 })
 
 const characterCount = $computed(() => {
-  let length = stringLength(htmlToText(editor.value?.getHTML() || ''))
+  const text = htmlToText(editor.value?.getHTML() || '')
+
+  let length = stringLength(text)
+
+  // taken from https://github.com/mastodon/mastodon/blob/07f8b4d1b19f734d04e69daeb4c3421ef9767aac/app/lib/text_formatter.rb
+  const linkRegex = /(https?:\/\/(www\.)?|xmpp:)\S+/g
+
+  // maximum of 23 chars per link
+  // https://github.com/elk-zone/elk/issues/1651
+  const maxLength = 23
+
+  length += [...text.matchAll(linkRegex)].reduce((sum, match) => sum - match[0].length + Math.min(maxLength, match[0].length), 0)
```

Comment:
can we use for...of to iterate over the matches once? It's a very slight perf improvement rather iterating over it twice.

Reply:
I could do that, but is it necessary in this case? The improvement should be negligible imo.

Reply:
it's also more readable and will be more maintainable with more contributors 🙏

Reply:
Done